### PR TITLE
Add a Function with error parameter to decide if repetition is necessary

### PR DESCRIPTION
**Pull request by ManAnRuck**
I added an optional parameter `doRetry` to decide if repetition is necessary.

doRetry takes an function with the error as parameter and returns a boolean

```typescript
(error: any) => boolean
```

### Example
```typescript
import { Retryable, BackOffPolicy } from './retry.decorator';

let count: number = 1;

class RetryExample {
  @Retryable({ 
    maxAttempts: 3,
    backOff: 1000,
    doRetry: (e: Error) => {
      return e.message === 'Error: 429';
    }
   })
  static async doRetry() {
    console.info(`Calling doRetry for the ${count++} time at ${new Date().toLocaleTimeString()}`);
    throw new Error('Error: 429');
  }

  @Retryable({ 
    maxAttempts: 3,
    backOff: 1000,
    doRetry: (e: Error) => {
      return e.message === 'Error: 429';
    }
   })
  static async doNotRetry() {
    console.info(`Calling doNotRetry for the ${count++} time at ${new Date().toLocaleTimeString()}`);
    throw new Error('Error: 404');
  }
}

(async () => {
  try {
    resetCount();
    await RetryExample.doRetry();
  } catch (e) {
    console.info(`All retry done as expected, final message: '${e.message}'`);
  }

  try {
    resetCount();
    await RetryExample.doNotRetry();
  } catch (e) {
    console.info(`All retry done as expected, final message: '${e.message}'`);
  }
  
})();

function resetCount() {
  count = 1;
}
```

Run the above code with `ts-node`, output will be:
```

Calling doRetry for the 1 time at 10:01:53 PM
Calling doRetry for the 2 time at 10:01:54 PM
Calling doRetry for the 3 time at 10:01:55 PM
Calling doRetry for the 4 time at 10:01:56 PM
All retry done as expected, final message: 'Failed for 'doRetry' for 3 times.'

Calling doNotRetry for the 1 time at 10:01:56 PM
All retry done as expected, final message: 'Error: 404'
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
   "name": "typescript-retry-decorator",
-  "version": "1.0.5",
+  "version": "1.1.11",
   "description": "A simple retry decorator for typescript with no dependency.",
   "main": "dist/index.js",
   "typings": "dist/index.d.ts",
   "scripts": {
     "test": "ts-node src/example.ts",
-    "build": "tsc -d -p ."
+    "build": "tsc",
```
Comment (vcfvct) [inline package.json]: Any reason you remove the `-d` which is for `.d.ts` file generation here?

File: src/retry.decorator.ts [hunk 2]
```diff
@@ -15,49 +15,58 @@ export function Retryable(options: RetryOptions): Function {
...
     }
-    descriptor.value = async function (...args: any[]) {
+    descriptor.value = async function(...args: any[]) {
       try {
-        return await retryAsync.apply(this, [originalFn, args, options.maxAttempts, options.backOff]);
+        return await retryAsync.apply(this, [originalFn, args, options.maxAttempts, options.backOff, options.doRetry]);
       } catch (e) {
-        e.message = `Failed for '${propertyKey}' for ${options.maxAttempts} times.`;
+        if (e.message === "maxAttempts") {
+          e.code = "429";
```
Comment (vcfvct) [inline src/retry.decorator.ts]: any reason you want an error code `429` here?

Comment (ManAnRuck) [reply src/retry.decorator.ts]: to continue handle this error it's more easy to compare by `code` and 429 (to many requests) is currently the most logic for me :)

File: package.json (hunk 1 above)
Comment (ManAnRuck) [reply package.json]: no, I change it back ;)

**Pull request merged**